### PR TITLE
[tests] Attempt to replicate sync issue in integration tests

### PR DESCRIPTION
**Pull request by delta1**
## Description

- Update the cucumber mining node process to accept `mine_on_tip_only` environment variable
- Fix some issues in mining node regarding max-blocks, min/max difficulty, and add some extra logging
- New integration test that attempts to reproduce a chain corruption issue when mining against a node that is still doing an initial sync

## Motivation and Context

Related to #2829 and issues found manually when mining against a syncing node

## How Has This Been Tested?

Manually and attempted cucumber test

## Types of changes

* [x] Bug fix (non-breaking change which fixes an issue)
* [ ] New feature (non-breaking change which adds functionality)
* [ ] Breaking change (fix or feature that would cause existing functionality to change)
* [ ] Feature refactor (No new feature or functional changes, but performance or technical debt improvements)
* [x] New Tests
* [ ] Documentation

## Checklist:

* [x] I'm merging against the `development` branch.
* [x] I ran `cargo-fmt --all` before pushing.
* [ ] I ran `cargo test` successfully before submitting my PR.
* [x] I have squashed my commits into a single commit.
* [ ] My change requires a change to the documentation.
* [ ] I have updated the documentation accordingly.
* [ ] I have added tests to cover my changes.

### CODE REVIEW

File: integration_tests/features/Sync.feature [hunk 1]
```diff
@@ -100,3 +100,20 @@ Feature: Block Sync
...
+    Given I have a seed node SEED
+    And I have wallet WALLET1 connected to seed node SEED
+    And I have wallet WALLET2 connected to seed node SEED
+    And I have mining node MINER connected to base node SEED and wallet WALLET1
+    And I have a base node SYNCER connected to all seed nodes
+    And I have mine-before-tip mining node MINER2 connected to base node SYNCER and wallet WALLET2
+    And I stop SYNCER
+    When I mine 1 blocks on SEED
+    Then node SEED is at height 1
+    When Mining node MINER mines 99 blocks with min difficulty 2 and max difficulty 100000
```
Comment (hansieodendaal) [inline integration_tests/features/Sync.feature:115]: **NIT:**
Proposal to make it easy to verify in the logs if the miner adhered to these limits
```
... with min difficulty 6 and max difficulty 20
```

File: integration_tests/features/Sync.feature [hunk 2]
```diff
@@ -100,3 +100,20 @@ Feature: Block Sync
...
+    Then node SEED is at height 100
+    When I start SYNCER
+    And Mining node MINER2 mines 5 blocks with min difficulty 1 and max difficulty 100000
```
Comment (hansieodendaal) [inline integration_tests/features/Sync.feature:118]: **NIT:**
Proposal to make it easy to verify in the logs if the miner adhered to these limits
```
... with min difficulty 1 and max difficulty 5
```

**Review by hansieodendaal — APPROVED** [review]
LGTM, tested, really nice. 

Just two nits, as I could not think of another way to confirm that difficulty limits are adhered to without unnecessary code changes.

File: integration_tests/features/Sync.feature (hunk 1 above)
Comment (delta1) [reply integration_tests/features/Sync.feature:115]: Thanks @hansieodendaal that's a good idea - do you mean in the mining node logs?

Comment (hansieodendaal) [reply integration_tests/features/Sync.feature:115]: Yes, it is there already, just easy to verify visually if clever limits are chosen.

**Pull request merged**